src/Code/DefinitionSummaryTooltip.elm

```diff
@@ -29,48 +30,73 @@ type DefinitionSummary
     | DataConstructorHover DataConstructorSummary
     | AbilityConstructorHover AbilityConstructorSummary
 
-
 type alias Model =
-    Maybe ( Reference, WebData DefinitionSummary )
+    {
+        data: Maybe ( Reference, WebData DefinitionSummary )
+        , alreadyFetched: Dict String (Reference, WebData DefinitionSummary)
+    }
 
 
 init : Model
 init =
-    Nothing
+    {
+        data = Nothing
+        , alreadyFetched = Dict.empty
+    }
 
 
 
 -- UPDATE
 
 
 type Msg
-    = ShowTooltip Reference
+    = ShowTooltipDelayed Reference
+    | ShowTooltip Reference
     | HideTooltip Reference
+    | CachedDefinition Reference (WebData DefinitionSummary)
     | FetchDefinitionFinished Reference (HttpResult DefinitionSummary)
 
-
 update : Config -> Msg -> Model -> ( Model, Cmd Msg )
 update config msg model =
+    let debounceDelay = 500
+    in
     case msg of
+        ShowTooltipDelayed ref ->
```

Comment:
What do you think about having this be `ShowTooltip` and the other one be `FetchDefinition`

Reply:
Yeah, that makes sense!